File: java-main/java-main/MagicSession/p7.java
class MaxProductSubarray {
    public static void main(String[] args) {
        int N = 5;
        int[] Arr = {6, -3, -10, 0, 2};

        int maxProduct = Arr[0];
        int minProduct = Arr[0];
        int result = Arr[0];

        for (int i = 1; i < N; i++) {
            if (Arr[i] < 0) {
                int temp = maxProduct;
                maxProduct = minProduct;
                minProduct = temp;
            }

            maxProduct = Math.max(Arr[i], maxProduct * Arr[i]);
            minProduct = Math.min(Arr[i], minProduct * Arr[i]);

            result = Math.max(result, maxProduct);
        }

        System.out.println("Maximum product subarray: " + result);
    }
}

File: bootcamp_c_cpp_dsa-main/bootcamp_c_cpp_dsa-main/C+DS/codes/Subarray/p7.c
// kadane's Algorithm


#include<stdio.h>
void main(){

	int arr[]={1,2,-3,-4,5,-6,7,-8,9};
	int sum=0;
	int max=0;

	for(int i=0i;i<10;i++){
		sum=sum+arr[i];
		if(sum<0)
			sum=0;
		if(max<sum)
			max=sum;
	}
	printf("max subarray sum=%d\n",max);
}
